app/controllers/application_controller.rb

```diff
@@ -468,7 +468,12 @@ def process_params_options(params)
     end
 
     options[:parent] = options[:parent] || @parent
-    options[:association] = HAS_ASSOCATION[params[:model_name]] if HAS_ASSOCATION.include?(params[:model_name])
+    if HAS_ASSOCATION.include?(params[:model])
+      options[:association] = HAS_ASSOCATION[params[:model]]
+    elsif options[:parent].kind_of?(GenericObject)
+      options.merge!(generate_options(params[:model]))
+    end
```

Comment:
I don't understand why this is needed. All options that are given to the GTL component are passwed throught to the `/report_data`.

Meaning we should not need to add any new exceptions here.

Actually I plan to remove the remaning exceptions and generalize this path.

Why don't you pass all the necessary options when rendering the grid?

Reply:
btw: even if we wanted to add th the exception here, I already wrote that `generate_options` is far too generic name in the context of the HUGE `ApplicationController`.